### PR TITLE
GUI-2247 Add alarm threshold lines to charts - POC

### PR DESCRIPTION
**Pull request by coyote240**
https://eucalyptus.atlassian.net/browse/GUI-2247

For tracer, POC, draw alarm threshold lines on the chart.

### CODE REVIEW

**Comment by euca-qa-gui-tester** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com/job/user-console-PR-monitor/1283/
Test FAILed.

**Comment by euca-qa-gui-tester** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com/job/user-console-PR-monitor/1284/
Test FAILed.

**Comment by coyote240** [comment]
Retest this, please.

**Comment by euca-qa-gui-tester** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com/job/user-console-PR-monitor/1287/
Test FAILed.

File: eucaconsole/static/js/pages/cloudwatch_charts.js [hunk 1]
```diff
@@ -10,6 +10,118 @@
...
 angular.module('CloudWatchCharts', ['EucaConsoleUtils'])
+.factory('CloudwatchAPI', ['$http', function ($http) {
+    return {
+        getChartData: function (params) {
+            return $http({
+                url: '/cloudwatch/api',
+                method: 'GET',
+                params: params
+            }).then(function (oData) {
+                if (typeof oData === 'string' && oData.indexOf('<html') > -1) {
```
Comment (dkavanagh) [inline eucaconsole/static/js/pages/cloudwatch_charts.js]: Probably ought to make use of common error handling from EucaConsoleUtils.

File: eucaconsole/static/js/pages/cloudwatch_charts.js [hunk 2]
```diff
@@ -10,6 +10,118 @@
...
+                return oData.data;
+            });
+        },
+
+        getAlarmsForMetric: function (metricName, params) {
+            return $http({
+                url: '/cloudwatch/alarms/json/' + metricName,
+                method: 'GET',
+                params: params
+            }).then(function (oData) {
```
Comment (dkavanagh) [inline eucaconsole/static/js/pages/cloudwatch_charts.js]: Could the functions be named to keep the intent clear? (success, error), like https://docs.angularjs.org/api/ng/service/$http

**Comment by euca-qa-gui-tester** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com/job/user-console-PR-monitor/1291/
Test PASSed.

**Comment by euca-qa-gui-tester** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com/job/user-console-PR-monitor/1301/
Test FAILed.

**Comment by coyote240** [comment]
Retest this, please.

**Comment by euca-qa-gui-tester** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com/job/user-console-PR-monitor/1302/
Test PASSed.

File: eucaconsole/static/js/pages/cloudwatch_charts.js [hunk 3]
```diff
@@ -10,6 +10,124 @@
...
+                params: params
+            }).then(function success (oData) {
+                if (typeof oData === 'string' && oData.indexOf('<html') > -1) {
+                    $('#timed-out-modal').foundation('reveal', 'open');
+                }
+                return oData.data;
+            }, function error (errorResponse) {
+                eucaHandleError(
+                    errorResponse.statusText,
+                    errorResponse.statuse);
```
Comment (kamalgill) [inline eucaconsole/static/js/pages/cloudwatch_charts.js]: @coyote240: is errorResponse.statuse a typo?  i.e. should it be errorResponse.status?

Comment (coyote240) [inline eucaconsole/static/js/pages/cloudwatch_charts.js]: Totally a type.

**Pull request merged**